searx/plugins/tor_check.py

```diff
@@ -0,0 +1,61 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+# lint: pylint
+"""
+A plugin to check if the ip address of the request is a TOR exit node
+if the user searches for "tor" "tor check" or "check tor"
+It fetches the tor exit node list from https://check.torproject.org/exit-addresses
+and parses all the IPs into a list, then checks if the user's ip address
+is in it.
+
+Enable in 'settings.yml' (uncomment "Tor check plugin"
+in the 'enabled_plugins' section)
+"""
+
+import re
+from urllib.request import urlopen
+from urllib.error import HTTPError
+
+name = "Tor check plugin"
+description = """
+This plugin checks if the address of the request is a TOR exit node,
+and informs the user if it is, like check.torproject.org but from searxng.
+"""
+default_on = False
+
+# Regex for exit node addresses in the list.
+reg = re.compile(r"(?<=ExitAddress )\S+")
+
+
+def post_search(request, search):
+
+    if search.search_query.query.lower() in ("tor", "tor check", "check tor"):
+
+        # Request the list of tor exit nodes.
+        try:
+            with urlopen("https://check.torproject.org/exit-addresses") as ex_nodes:
+                node_list = re.findall(reg, str(ex_nodes.read()))
```

Comment:
```python
resp = get("https://check.torproject.org/exit-addresses")
node_list = re.findall(reg, str(resp.text))
```

Just to make sure: does it makes sense to cache the response?
It seems no looking how often the content change ( https://collector.torproject.org/recent/exit-lists/ )

Reply:
I don't really know what else can i do.
Do you mean something like:
`node_list = re.findall(reg, resp = get("https://check.torproject.org/exit-addresses"))` ?
Or 
`if ip_address in re.findall(reg, resp = get("https://check.torproject.org/exit-addresses")):` ?

Reply:
> Just to make sure: does it makes sense to cache the response?

I don't know a rule for, I leave it un-cached  for its first / when we have a rule to update the cache, we can implement a new PR.